### ISSUE TITLE
Script button or F3should open external editor if one is being used

### ISSUE THREAD

**Issue by DJ-Dav** [godotengine/godot-proposals#2590]
### Describe the project you are working on
I am making a game with a complicated UI using C#, not GDscript.

### Describe the problem or limitation you are having in your project
If I use the layout button above the viewport and then I want to write some code, I click the script tab, which opens the built-in editor, not the external editor I am using.

### Describe the feature/enhancement and how it helps to overcome the problem or limitation
If the button opened the external editor tab, it could save a few needlessly wasted seconds, which across the development process can add up to an unreasonable amount.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams
if script_button_pressed and settings.external_editor == true:
    launch_external_editor()

I don't know what to call the functions and variables, so I wrote this simple pseudo-code instead.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

I'm sure that many people use external editors for code, so having the script tab at the top open the built-in editor is rather useless to quite a few people.

### Is there a reason why this should be core and not an add-on in the asset library?
Adding an add-on is more time wasted than having it as default functionality.

**Comment by pycbouh** [comment]
That doesn't make much practical sense. That button is specifically to open the built-in editor, and this change would make it completely inaccessible. You also won't be able to return to the Godot editor with other related hot keys, so it would make this a one way only journey.

**Comment by DJ-Dav** [comment]
It's not really a duplicate, because even using a supported editor it's a problem, if I have chosen an external editor I have no need for the built in one, so I think enabling an external editor should change the functionality of the top bar button, even if this is feature in itself is a setting.

**Comment by DJ-Dav** [comment]
And just to clarify,ifno external editor is specified, the button would serve its default purpose, the built in editor wouldn't be inaccessible, just replaced if a certain setting is chosen.

**Comment by pycbouh** [comment]
> And just to clarify,ifno external editor is specified, the button would serve its default purpose, the built in editor wouldn't be inaccessible, just replaced if a certain setting is chosen.

Yes, my point is that it will be completely inaccessible unless you disable your external editor. Which may not be ideal, at least not as the default or only behavior.

> It's not really a duplicate, because even using a supported editor it's a problem

It doesn't matter, the core of both issues, to me, is the same.

**Comment by rick551a** [comment]
Projects can have a mix of script types (e.g. GDScript & c#) so having a shortcut to both is sometimes needed.
